### PR TITLE
Functions gpg output and doc

### PR DESCRIPTION
**Pull request by evelikov**
So I've tried using repro and due to some gpg server glitch on my end, I wasn't able to use it.
Turns out the gpg output was completely discarded :-\

While looking at reinstating that, I've noticed an "issue" after another, so here's a lovely series which:
 - removes unneeded statements
 - moves code where it's applicable
 - reverts the gpg redirect, now that it happens once
 - reshuffles the arg parsing, making it more robust and improving docs
 - factors a handful of functions for clarity (and sanity) sake

### CODE REVIEW

File: docs/repro.8.txt [hunk 1]
```diff
@@ -37,6 +37,9 @@ Options
         directory. This one is used to build the package instead of checking out
         a PKGBUILD from asp.
 
+ *-n*::
+        When enable,d `repro` will run makepkg with --nocheck.
```
Comment (jelly) [inline docs/repro.8.txt]: typo: enabled

Comment (evelikov) [reply docs/repro.8.txt]: Thanks fixed locally. Will respin/submit one the NOCHECK is merged.

**Comment by evelikov** [comment]
Rebased resolving a doc and OUTDIR merge conflicts, while *changing the last patch a to simple colorize move (instead of main function).

**Comment by evelikov** [comment]
Rebased on top of the `makepkg_args` change.

**Comment by evelikov** [comment]
Gentle poke.

**Comment by Foxboron** [comment]
Thanks! Sorry for the delay

**Pull request merged**